File: CS/SecuredExportExample.Module/BusinessObjects/Employee.cs
using System;
using System.ComponentModel;

using DevExpress.Xpo;
using DevExpress.Data.Filtering;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.BaseImpl.PermissionPolicy;
using DevExpress.Persistent.Validation;
using DevExpress.ExpressApp.Security;

namespace SecuredExportExample.Module.BusinessObjects {
    [DefaultClassOptions, ImageName("BO_Employee")]
    public class Employee : PermissionPolicyUser {
        public Employee(Session session)
            : base(session) { }
        [Association("Employee-Task")]
        public XPCollection<Task> Tasks {
            get { return GetCollection<Task>("Tasks");  }
        }
        
    }
    [DefaultClassOptions]
    [ImageName("BO_Task")]
    public class Task : BaseObject {
        public Task(Session session)
            : base(session) { }
        private string subject;
        public string Subject {
            get { return subject; }
            set { SetPropertyValue("Subject", ref subject, value); }
        }
        private DateTime dueDate;
        public DateTime DueDate {
            get { return dueDate; }
            set { SetPropertyValue("DueDate", ref dueDate, value); }
        }
        private Employee assignedTo;
        [Association("Employee-Task")]
        public Employee AssignedTo {
            get { return assignedTo; }
            set { SetPropertyValue("AssignedTo", ref assignedTo, value); }
        }
    }
}

File: VB/SecuredExportExample.Module/BusinessObjects/Employee.vb
Imports System
Imports System.ComponentModel

Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.BaseImpl.PermissionPolicy
Imports DevExpress.Persistent.Validation
Imports DevExpress.ExpressApp.Security

Namespace SecuredExportExample.Module.BusinessObjects
    <DefaultClassOptions, ImageName("BO_Employee")> _
    Public Class Employee
        Inherits PermissionPolicyUser

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
        <Association("Employee-Task")> _
        Public ReadOnly Property Tasks() As XPCollection(Of Task)
            Get
                Return GetCollection(Of Task)("Tasks")
            End Get
        End Property

    End Class
    <DefaultClassOptions, ImageName("BO_Task")> _
    Public Class Task
        Inherits BaseObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Private subject_Renamed As String
        Public Property Subject() As String
            Get
                Return subject_Renamed
            End Get
            Set(ByVal value As String)
                SetPropertyValue("Subject", subject_Renamed, value)
            End Set
        End Property

        Private dueDate_Renamed As Date
        Public Property DueDate() As Date
            Get
                Return dueDate_Renamed
            End Get
            Set(ByVal value As Date)
                SetPropertyValue("DueDate", dueDate_Renamed, value)
            End Set
        End Property

        Private assignedTo_Renamed As Employee
        <Association("Employee-Task")> _
        Public Property AssignedTo() As Employee
            Get
                Return assignedTo_Renamed
            End Get
            Set(ByVal value As Employee)
                SetPropertyValue("AssignedTo", assignedTo_Renamed, value)
            End Set
        End Property
    End Class
End Namespace
